td-shim-tools/src/enroller.rs

```diff
@@ -32,8 +32,12 @@ pub fn build_cfv_header() -> FvHeader {
         .fv_header
         .file_system_guid
         .copy_from_slice(FIRMWARE_FILE_SYSTEM3_GUID.as_bytes());
+    cfv_header.fv_header.signature = FVH_SIGNATURE;
+    cfv_header.fv_header.header_length = size_of::<FvHeader>() as u16;
     cfv_header.fv_header.fv_length = TD_SHIM_CONFIG_SIZE as u64;
-    cfv_header.fv_header.checksum = 0xdc0a;
+    cfv_header.fv_header.revision = FVH_REVISION;
+    cfv_header.fv_header.update_checksum();
```

Comment:
Calculating the checksum looks like an improvement to me but see my comments on this method below.